### PR TITLE
great idea, terrible execution. 

### PR DESCRIPTION
**Pull request by avavita**

### CODE REVIEW

File: triangle_facts/triangle_facts.rb [hunk 1]
```diff
@@ -1,58 +1,26 @@
...
-	end
+class TriangleFacts
+  require_relative 'triangle'
+
+  def self.recite_facts(triangle)
+    if triangle.equilateral
+      puts "This triangle is equilateral!"
+    elsif triangle.isosceles
+      puts "This triangle is isosceles! Also, that word is hard to type."
+    elsif triangle.scalene
```
Comment (djkotowski) [inline triangle_facts/triangle_facts.rb]: If a triangle is neither equilateral nor isosceles, it is scalene.

File: triangle_facts/triangle.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  def isosceles
+    [side1,side2,side3].uniq.length == 2
+  end
+
+  def scalene
+    !equilateral && !isosceles
+  end
+
+  def right
+    angles.include? 90
```
Comment (djkotowski) [inline triangle_facts/triangle.rb]: Wrap args in parentheses.

File: triangle_facts/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  def radians
+  [
+    Math.acos((side2**2 + side3**2 - side1**2) / (2.0 * side2 * side3)),
+    Math.acos((side1**2 + side3**2 - side2**2) / (2.0 * side1 * side3)),
+    Math.acos((side1**2 + side2**2 - side3**2) / (2.0 * side1 * side2))
+  ]
+  end
+
+  private
+  def radians_to_degrees(rads)
```
Comment (djkotowski) [inline triangle_facts/triangle.rb]: Put a blank line before _and_ after `private`.

File: triangle_facts/triangle.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def angles
+    rads = radians #avoiding running the method 3 times
+
+    angle1 = radians_to_degrees(rads[0])
+    angle2 = radians_to_degrees(rads[1])
+    angle3 = radians_to_degrees(rads[2])
+
+    [angle1, angle2, angle3]
```
Comment (djkotowski) [inline triangle_facts/triangle.rb]: You could also shorten this to a single line using `#map`.

File: triangle_facts/triangle.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+class Triangle
+  attr_accessor :side1,:side2,:side3
+
+  def initialize(side1,side2,side3)
+    @side1 = side1
+    @side2 = side2
+    @side3 = side3
+  end
+
+  def equilateral
```
Comment (djkotowski) [inline triangle_facts/triangle.rb]: Predicate methods should end with a `?`

File: triangle_facts/triangle.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    side1 == side2 && side2 == side3
+  end
+
+  def isosceles
```
Comment (djkotowski) [inline triangle_facts/triangle.rb]: Ditto above.

**Comment by djkotowski** [comment]
Looks good.  No need for a resubmission.  Just a few comments throughout.

File: triangle_facts/triangle.rb (hunk 3 above)
Comment (avavita) [inline triangle_facts/triangle.rb]: Can you link me the style guide that most people refer to? The github style guide disagrees here.

Comment (avavita) [inline triangle_facts/triangle.rb]: also, is something like this okay?
```ruby
  def generate_char
    ('A'..'Z').to_a.sample 
  end

  def generate_num
    rand(10)
  end

  private :generate_char, :generate_num
```

File: triangle_facts/triangle.rb (hunk 2 above)
Comment (kotp) [inline triangle_facts/triangle.rb]: Wrap _some_ arguments in parenthesis, it is uncommon to wrap all arguments in parenthesis.

For instance, `length.==(2)` looks strange, as well as `side1.==(side2)` though technically correct.

File: triangle_facts/triangle.rb (hunk 3 above)
Comment (djkotowski) [inline triangle_facts/triangle.rb]: We prefer not using that syntax, as it is not evident until you reach the `private` line that those methods are private.  By including it before the methods, it becomes more apparent.

Comment (djkotowski) [inline triangle_facts/triangle.rb]: Our primary style guide is the [Thoughtbot Style Guide](https://github.com/thoughtbot/guides/tree/master/style#ruby).  The blank line below `private` is mostly an in-house guideline.
